packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_basic_auth_request.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Defines the parameters of a pending HTTP basic authentication request.
+class HttpBasicAuthRequest {
+  /// Creates a [HttpBasicAuthRequest].
+  const HttpBasicAuthRequest({
+    required this.onProceed,
+    required this.onCancel,
+    required this.host,
+    required this.realm,
+  });
+
+  /// The callback to authenticate.
+  final void Function(String username, String password) onProceed;
```

Comment:
It looks like iOS can include more than just the `username` and `password` when proceeding (e.g. `NSURLCrendtial.persistence`). I think we can change the method to take a single parameter, something like `WebViewCredential`, and add the username and password to this class. Then we can add additional parameters with an iOS implementation in a future update.